### PR TITLE
feat: edit validator page

### PR DESCRIPTION
**Pull request by taro04**
WIP
- edit-simple
image
- edit
image

### CODE REVIEW

**Comment by YasunoriMATSUOKA** [comment]
Sorry, I did not ask you, but please implement the edit-validator/multiple page as well, referring to the create-validator/multiple page.

File: projects/portal/src/app/models/cosmos/staking.service.ts [hunk 1]
```diff
@@ -182,6 +182,148 @@ export class StakingService {
...
+      */
+      commission_rate: '0.10',
+      min_self_delegation: editValidatorData.min_self_delegation,
+      /*
+      value: {
+        denom: editValidatorData.denom,
+        amount: editValidatorData.amount,
+      },
+      */
+    };
```
Comment (taro04) [inline projects/portal/src/app/models/cosmos/staking.service.ts:291]: Is this "editValidatorTxData" OK?
it comes from "proto.d.ts"
( commisson_rate is on debug )
image

Comment (YasunoriMATSUOKA) [reply projects/portal/src/app/models/cosmos/staking.service.ts:291]: 1. I guess you should refer cosmos.staking.v1beta1.IMsgEditValidator.
2. I guess you need to set commission_rate query parameter to be 0.10 like value and you need to convert it to "100000000000000000". this is strange but it seems 10% => 0.1 => 0.1 * 10^18 maybe that seems 18digits fixed style string.

**Comment by taro04** [comment]
How do I create the query parameter "user ID" that edit-validator / multi-page is using?

File: projects/portal/src/app/models/cosmos/staking.application.service.ts [hunk 2]
```diff
@@ -270,6 +271,193 @@ export class StakingApplicationService {
...
+      const txHash = editValidatorResult.tx_response?.txhash;
+      if (txHash === undefined) {
+        throw Error('Invalid txHash!');
+      }
+      this.snackBar.open('Success', undefined, { duration: 6000 });
+      if (!disableRedirect) {
+        const redirectUrl =
+          location.port === '80' || location.port === '443' || location.port === ''
+            ? `${location.protocol}//${location.hostname}/explorer/validators/${editValidatorData.validator_address}`
+            : `${location.protocol}//${location.host}/explorer/validators/${editValidatorData.validator_address}`;
```
Comment (taro04) [inline projects/portal/src/app/models/cosmos/staking.application.service.ts:357]: This implementation was brought from createValidator.
Why do you redirect to the same port for ports 443 and 80?
I think `${location.protocol}//${location.host}/explorer/validators/${editValidatorData.validator_address};` is enough.

File: projects/portal/src/app/models/cosmos/staking.service.ts [hunk 3]
```diff
@@ -182,6 +182,148 @@ export class StakingService {
...
+      },
+    });
+    console.log('debug_1-3');
+
+    // sign
+    const txBuilder = new cosmosclient.TxBuilder(sdk, txBody, authInfo);
+    //const signDocBytes = txBuilder.signDocBytes(baseAccount.account_number);
+    //txBuilder.addSignature(privKey.sign(signDocBytes));
+    console.log('debug_1-4', txBuilder);
+
```
[the remark this answers is not in the document]
Comment (taro04) [reply projects/portal/src/app/models/cosmos/staking.service.ts:288]: - Simulation is fail, but the actual tx announce is sucess like createValidator, 
- I referred to createDeligate. (Simulation is unsigned. Announce is signed.)

File: projects/portal/src/app/models/cosmos/staking.application.service.ts [hunk 4]
```diff
@@ -270,6 +270,193 @@ export class StakingApplicationService {
...
+      this.snackBar.open('Failed to get Wallet info from dialog! Tray again!', 'Close');
+      return;
+    }
+
+    if (!validatePrivateStoredWallet(privateWallet)) {
+      this.snackBar.open('Invalid Wallet info!', 'Close');
+      return;
+    }
+
+    const currentCosmosWallet = await this.walletService.currentCosmosWallet$
```
Comment (taro04) [inline projects/portal/src/app/models/cosmos/staking.application.service.ts:389]: I made this `currentCosmosWallet` for common methods of `signTx` in `staking.editValidator()`
Is it OK? 
Should I somehow convent `privateWallet: StoredWallet` (Line 377) to `currentCosmosWallet`

**Comment by YasunoriMATSUOKA** [comment]
> How do I create the query parameter "user ID" that edit-validator / multi-page is using?

We(You and I) can't create userID because it is created by Neukind's User's Dashboard application.
For multiple page, current implementation will be enough.
It will be implemented or fixed by Neukind's developer if necessary.

File: projects/portal/src/app/models/cosmos/staking.service.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (YasunoriMATSUOKA) [reply projects/portal/src/app/models/cosmos/staking.service.ts:288]: It seems fine. Thank you!

File: projects/portal/src/app/models/cosmos/staking.application.service.ts (hunk 2 above)
Comment (YasunoriMATSUOKA) [reply projects/portal/src/app/models/cosmos/staking.application.service.ts:356]: I guess you are right. This is redundant and confusing. Can you fix it with your simple way? Thank you!

**Comment by taro04** [comment]
@YasunoriMATSUOKA 
Thank you. This is the URL.

- edit-validator
[node-1]
[http://127.0.0.1:4200/portal/staking/edit-validator?moniker=test-1X&identity=test-1X&website=https%3A%2F%2Fcauchye.com&security_contact=info%40cauchye.com&details=This%20is%20a%20validator%20node%20to%20test%20MsgCreateValidator%20tx.&rate=10&min_self_delegation=1&delegator_address=ununifi1wpcx4k3v9juxjqvvtqtguprv2axng4tyt2tf6e](http://127.0.0.1:4200/portal/staking/edit-validator?moniker=test-1X&identity=test-1X&website=https%3A%2F%2Fcauchye.com&security_contact=info%40cauchye.com&details=This%20is%20a%20validator%20node%20to%20test%20MsgCreateValidator%20tx.&rate=10&min_self_delegation=1&delegator_address=ununifi1wpcx4k3v9juxjqvvtqtguprv2axng4tyt2tf6e)
[node-2]
[http://127.0.0.1:4200/portal/staking/edit-validator?moniker=test-2X&identity=test-2X&website=https%3A%2F%2Fcauchye.com&security_contact=info%40cauchye.com&details=debug_check_edit_validator&rate=11&min_self_delegation=2&delegator_address=ununifi1qjkv2yns0amduey6umvtxkpqllj9dkrpfc5dw4](http://127.0.0.1:4200/portal/staking/edit-validator?moniker=test-2X&identity=test-2X&website=https%3A%2F%2Fcauchye.com&security_contact=info%40cauchye.com&details=debug_check_edit_validator&rate=11&min_self_delegation=2&delegator_address=ununifi1qjkv2yns0amduey6umvtxkpqllj9dkrpfc5dw4)

- edit-validator-simple
[node-6]
[http://127.0.0.1:4200/portal/staking/edit-validator/simple?moniker=node-6edit&identity=node-6X&website=https%3A%2F%2Fcauchye.com6X&security_contact=edit6%40cauchye.com&details=edit%20by%20edit%20page%20node6&rate=10&min_self_delegation=2&delegator_address=ununifi1y42j8fgmz2uj8dyrdxxj095kyn8r5lscsj2qg4](http://127.0.0.1:4200/portal/staking/edit-validator/simple?moniker=node-6edit&identity=node-6X&website=https%3A%2F%2Fcauchye.com6X&security_contact=edit6%40cauchye.com&details=edit%20by%20edit%20page%20node6&rate=10&min_self_delegation=2&delegator_address=ununifi1y42j8fgmz2uj8dyrdxxj095kyn8r5lscsj2qg4)
[node-7]
[http://127.0.0.1:4200/portal/staking/edit-validator/simple?moniker=node-7XX&identity=node-7XX&website=https%3A%2F%2Fcauchye.com6X&security_contact=edit7%40cauchye.com&details=edit%20by%20edit%20page%20node7&rate=10&min_self_delegation=2&delegator_address=ununifi1nxjw30k9g3fawtnnrw57f0h48h5rdclkrl0kez](http://127.0.0.1:4200/portal/staking/edit-validator/simple?moniker=node-7XX&identity=node-7XX&website=https%3A%2F%2Fcauchye.com6X&security_contact=edit7%40cauchye.com&details=edit%20by%20edit%20page%20node7&rate=10&min_self_delegation=2&delegator_address=ununifi1nxjw30k9g3fawtnnrw57f0h48h5rdclkrl0kez)

File: projects/portal/src/assets/config.js [hunk 5]
```diff
@@ -6,13 +6,13 @@ const faucetJpuPort = location.protocol === 'https:' ? 8005 : 8004;
...
-const domainCauchyED = 'd.private-test.ununifi.cauchye.net';
+const domainCauchyEA = '45.76.96.57';
+const domainCauchyEB = '104.238.151.213';
+const domainCauchyEC = '45.63.121.67';
+const domainCauchyED = '207.148.89.195';
 
-const chainID = 'ununifi-8-private-test';
-const chainName = 'UnUnuFi (Private)';
+const chainID = 'create-validator-debug';
+const chainName = 'UnUnuFi (create-validator-debug)';
```
Comment (taro04) [inline projects/portal/src/assets/config.js:16]: Should I not modify this file?
( This Chain "create-validator-debug" is only for debug. )

File: projects/portal/src/app/models/cosmos/staking.application.service.ts (hunk 4 above)
Comment (taro04) [reply projects/portal/src/app/models/cosmos/staking.application.service.ts]: I used `convertStoredWalletToCosmosWallet` functon in  waller.service.ts

File: projects/portal/src/assets/config.js (hunk 5 above)
Comment (YasunoriMATSUOKA) [reply projects/portal/src/assets/config.js:16]: As you said, This "create-validator-debug" is only for debug, I wanna ask you to put it back. I will debug with appropriate network settings.

Comment (taro04) [reply projects/portal/src/assets/config.js]: Reverted config.js changes.

**Comment by taro04** [comment]
@YasunoriMATSUOKA 

Sorry, I didn't notice your repry.
Reverted config.js changes.

File: projects/portal/src/app/views/staking/edit-validator-simple/edit-validator-simple.component.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+<form class="h-screen" #formRef="ngForm">
+  <input
+    #fileInputRef
+    name="fileInput"
+    type="file"
+    multiple="false"
+    accept="image/*"
+    (change)="onChangeFile($event)"
+  />
+</form>
```
Comment (KimuraYu45z) [inline projects/portal/src/app/views/staking/edit-validator-simple/edit-validator-simple.component.html:11]: You can edit the design of file input button by two steps:

- Set the visibility of file input button as `hidden`
- Create simple normal button with onClick handler. If it is clicked, JS code for emitting click event of **file input button** will be evoked.

Comment (taro04) [reply projects/portal/src/app/views/staking/edit-validator-simple/edit-validator-simple.component.html:11]: Thank you for your comment.
Done as below.
image

**Comment by taro04** [comment]
@YasunoriMATSUOKA 
Thank you for your review.
I resolved conflict and delete link to "edit-validator-multiple"
Please review.

**Review by YasunoriMATSUOKA — APPROVED** [review]
LGTM

**Pull request merged**